dronekit/__init__.py

```diff
@@ -1007,6 +1008,12 @@ def listener(_, name, value):
 
         @handler.forward_message
         def listener(_, msg):
+            if self._target_system is None:
+                return
+            if self._target_system:
+                src = msg.get_srcSystem()
+                if src != self._target_system and src != 0:
+                    return
```

Comment:
On Tue, 12 Apr 2016, Ramon Roche wrote:

> should we deal with this at the forward_message level?

Further patches in this PR move this code into mavlink.py's MAVSystem 
object - so, yes, sort of :-)